apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,205 @@
         res.locals.authn_user.user_id,
-        function (err, job_sequence_id) {
-          if (ERR(err, next)) return;
-          res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
-        },
+        close,
+        overrideGradeRate,
       );
-    });
-  } else if (req.body.__action === 'set_time_limit') {
-    const params = {
-      assessment_instance_id: req.body.assessment_instance_id,
-      assessment_id: res.locals.assessment.id,
-      time_add: req.body.time_add,
-      time_ref: req.body.time_ref,
-      base_time: 'date_limit',
-      authn_user_id: res.locals.authz_data.authn_user.user_id,
-    };
-    if (req.body.plus_minus === 'unlimited') {
-      params.base_time = 'null';
-    } else if (req.body.plus_minus === 'expire') {
-      params.base_time = 'current_date';
-      params.time_add = 0;
-      params.time_ref = 'minutes';
-    } else if (req.body.plus_minus === 'set_total') {
-      params.base_time = 'start_date';
-    } else if (req.body.plus_minus === 'set_rem') {
-      params.base_time = 'current_date';
-    } else {
-      params.time_add *= req.body.plus_minus;
-    }
-    sqldb.query(sql.set_time_limit, params, function (err) {
-      if (ERR(err, next)) return;
+      res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
+    } else if (req.body.__action === 'delete_all') {
+      await sqldb.callAsync('assessment_instances_delete_all', [
+        res.locals.assessment.id,
+        res.locals.authn_user.user_id,
+      ]);
       res.send(JSON.stringify({}));
-    });
-  } else if (req.body.__action === 'set_time_limit_all') {
-    const params = {
-      assessment_id: res.locals.assessment.id,
-      time_add: req.body.time_add,
-      time_ref: req.body.time_ref,
-      base_time: 'date_limit',
-      reopen_closed: !!req.body.reopen_closed,
-      authn_user_id: res.locals.authz_data.authn_user.user_id,
-    };
-    if (req.body.plus_minus === 'unlimited') {
-      params.base_time = 'null';
-    } else if (req.body.plus_minus === 'expire') {
-      params.base_time = 'current_date';
-      params.time_add = 0;
-      params.time_ref = 'minutes';
-    } else if (req.body.plus_minus === 'set_total') {
-      params.base_time = 'start_date';
-    } else if (req.body.plus_minus === 'set_rem') {
-      params.base_time = 'current_date';
-    } else {
-      params.time_add *= req.body.plus_minus;
-    }
-    sqldb.query(sql.set_time_limit_all, params, function (err) {
-      if (ERR(err, next)) return;
+    } else if (req.body.__action === 'regrade') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      await checkBelongsAsync(assessment_instance_id, assessment_id);
+      const job_sequence_id = await regradeAssessmentInstance(
+        assessment_instance_id,
+        res.locals.user.user_id,
+        res.locals.authn_user.user_id,
+      );
+      console.log(job_sequence_id);
```

Comment:
This should be deleted.

Reply:
Deleted in commit [ecafed4](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/ecafed4ec68b0006158dd7f2866a9331734aa370).